### PR TITLE
CAPZ: add k8s ci artifacts conformance periodic

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
Following https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1018, updates the Cluster API Azure periodic jobs to use the new framework with CI artifacts for testing conformance against k8s master.

/hold for https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1018

/assign @nader-ziada @cpanato

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics.yaml [hunk 1]
```diff
@@ -19,9 +19,6 @@ periodics:
         command:
         - "runner.sh"
         - "./scripts/ci-conformance.sh"
-        env:
-          - name: GINKGO_FOCUS
-            value: "Conformance Tests"
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-periodics.yaml:24]: not needed anymore https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1018/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R188

File: config/jobs/kubernetes/sig-cloud-provider/azure/sig-azure-config.yaml [hunk 2]
```diff
@@ -271,95 +271,3 @@ periodics:
     testgrid-alert-email: kubernetes-provider-azure@googlegroups.com
     description: Runs conformance & node conformance tests latest kubernetes master with aks-engine
     testgrid-num-columns-recent: '30'
-- interval: 24h
-  name: ci-kubernetes-e2e-capz-conformance-master
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes/sig-cloud-provider/azure/sig-azure-config.yaml:275]: replaced by periodic-cluster-api-provider-azure-conformance-with-ci-artifacts

new job added to the same dashboards

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by CecileRobertMichon** [comment]
rebased 

@chewong @cpanato PTAL

**Review by cpanato — APPROVED** [review]
/lgtm

**Review by chewong — APPROVED** [review]
/lgtm
/approve
for `config/jobs/kubernetes/sig-cloud-provider/azure/sig-azure-config.yaml`

**Pull request merged**